rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/facade/compilerDirectives.kt

```diff
@@ -220,14 +220,11 @@ internal fun String.resolveCompilerDirectives(): String {
 }
 
 private fun String.transformToComment(): String {
-    val newString: String
-    if (this.length >= 7) {
-        newString = this.substring(0, 6) + '*' + this.substring(7)
-    } else {
-        val padding = "       "
-        newString = this + padding.substring(0, 7 - this.length) + '*'
+    return when {
+        this.isBlank() -> "     *"
```

Comment:
"There is a missing space; the asterisk (*) must be placed in column 7. Even if your fix works, it might just be because the parser isn't perfect.

I discovered this potential issue because I didn't understand the cause of the original error. Then I modified your test to print the source passed to the parser:

```kotlin
/**
 * Duplicate definition in two different copies
 * @see #LS24004074
 */
@Test
fun executeDUPDEF() {
    val expected = listOf("ok")
    // I implemented the callback to print the source code before parsing
    val configuration = Configuration().apply {
        this.jarikoCallback.beforeParsing = { source ->
            println("source:")
            println(source)
            source
        }
    }
    assertEquals(expected, "compilerDirectives/DUPDEF".outputOf(configuration = configuration))
}
```

I hope this trick will be useful in the future."

Reply:
I updated the implementation changing also the `else` branch that appeared to suffer from the opposite issue (overshooting and placing the `*` in 8th position).